newlib/libc/include/complex.h

```diff
@@ -112,9 +112,9 @@ float crealf(float complex);
 
 #if __ISO_C_VISIBLE >= 2011
 #ifdef _HAVE_BUILTIN_COMPLEX
-#define CMPLX(r,i) __builtin_complex(r,i)
-#define CMPLXF(r,i) __builtin_complex(r,i)
-#define CMPLXL(r,i) __builtin_complex(r,i)
+#define CMPLX(r,i) __builtin_complex((double)r, (double)i)
```

Comment:
The macro argument names need to be enclosed in parenthesis to ensure the expected evaluation order. So,
`#define COMPLX(r,i) __builtin_complex((double)(r), (double)(i))`

Reply:
@keith-packard I have added the parentheses, and I think the PR is ready to be merged for now.